src/resolve/resolved-dns-packet.c

```diff
@@ -47,13 +50,14 @@ int dns_packet_new(DnsPacket **ret, DnsProtocol protocol, size_t mtu) {
 
         assert(ret);
 
-        if (mtu <= UDP_PACKET_HEADER_SIZE)
+        /* When dns_packet_new() is called with mtu == 0, allocate more than the
+         * absolute minimum (which is the dns packet header size), to avoid
+         * resizing immediately again after appending the first data to the packet.
+         */
+        if (mtu < UDP_PACKET_HEADER_SIZE)
```

Comment:
@keszybz @poettering  I was trying to understand what dns_packet_new() is really doing, and I think it may have still an error.
Why the size_t argument is named mtu ? This is not the source of the confusion ?
The only call to dns_packet_new() with a non null size is in manager_recv() of resolved-manager.c

It looks like the size pass is just the size of the data, without the UDP header, so why UDP_PACKET_HEADER_SIZE is used in the current computation ?